src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -337,312 +371,212 @@
-
-        &.datagrid-empty {
-            border-top: 0;
-            padding: 0.5rem;
-            display: flex;
-            flex-flow: column nowrap;
-            align-items: center;
-            justify-content: center;
-            font-size: clr-getTypePropertyValueForDomElement(datagrid_placeholder, font-size);
-            color: $clr-dark-midtone-gray;
+            &.datagrid-empty {
+                border-top: 0;
+                padding: 0.5rem;
+                display: flex;
+                flex-flow: column nowrap;
+                align-items: center;
+                justify-content: center;
+                font-size: clr-getTypePropertyValueForDomElement(datagrid_placeholder, font-size);
+                color: $clr-dark-midtone-gray;
 
-            .datagrid-placeholder-image {
-                width: 2.5rem;
-                height: 2.5rem;
-                margin-bottom: 0.5rem;
-                background-repeat: no-repeat;
-                background-size: contain;
-                background-position: center;
-                background-image: generateEmptyDatagridPlaceholder();
+                .datagrid-placeholder-image {
+                    width: 2.5rem;
+                    height: 2.5rem;
+                    margin-bottom: 0.5rem;
+                    background-repeat: no-repeat;
+                    background-size: contain;
+                    background-position: center;
+                    background-image: generateEmptyDatagridPlaceholder();
+                }
             }
         }
     }
 
-    .datagrid-action-bar {
-        flex: 0 0 auto;
-        display: flex;
-        flex-flow: row nowrap;
-        align-items: stretch;
-
-        // Datagrid has a margin top of 1 baseline. action bar is supposed to be 1/2 baseline from the datagrid
-        // Had to use translate instead of negative margins to pull the
-        // action bar down. negative margin screws up the click targets.
-        transform: translateY(0.5rem);
-    }
-
-    .datagrid-foot {
+    .datagrid-footer {
         flex: 0 0 auto;
         display: flex;
         flex-flow: row nowrap;
-        justify-content: flex-end;
-        align-items: stretch;
         height: 1.5rem;
         padding: 0 $clr-table-cellpadding;
         // Account for borders
         line-height: calc(1.5rem - #{3 * $clr-default-borderwidth});
         font-size: clr-getTypePropertyValueForDomElement(table_header, font-size);
         background-color: $clr-thead-bgcolor;
-        border-top: (2 * $clr-default-borderwidth) solid $clr-table-footer-border-top-color;
+        border-top: $clr-default-borderwidth solid $clr-table-footer-border-top-color;
         border-radius: 0 0 $clr-global-borderradius $clr-global-borderradius;
+        border-left: $clr-table-borderstyle;
+        border-bottom: $clr-table-borderstyle;
+        border-right: $clr-table-borderstyle;
+
+        .datagrid-footer-left, .datagrid-footer-right {
+            flex: 1;
```

Comment:
See my previous PR on the footer, I'm positive you want the `.datagrid-footer-left` to _not_ be flexible (`flex: 0 0 auto)`) and the `footer-right` to be the one filling up the space.